src/hooks/useInfiniteScroll.ts

```diff
@@ -0,0 +1,42 @@
+import { useCallback, useEffect, useRef } from 'react';
+
+interface UseInfiniteScrollProps {
+  hasNextPage: boolean;
+  isFetchingNextPage: boolean;
+  fetchNextPage: () => void;
+}
+
+export default function useInfiniteScroll({
+  hasNextPage,
+  isFetchingNextPage,
+  fetchNextPage,
+}: UseInfiniteScrollProps) {
+  const observerRef = useRef<IntersectionObserver | null>(null);
```

Comment:
```suggestion
  const observerRef = useRef<IntersectionObserver>(null);
```
💬 이렇게 써도 됐던 것으로 기억합니다!

Reply:
오 감사합니다!